src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DotApps, DotAppsSaveData } from '@shared/models/dot-apps/dot-apps.model';
+import { ActivatedRoute } from '@angular/router';
+import { pluck, take } from 'rxjs/operators';
+
+import { AppsResolverData } from '../dot-apps-configuration/dot-apps-configuration-resolver.service';
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+import { DotAppsService } from '@services/dot-apps/dot-apps.service';
+import { DotAlertConfirmService } from '@services/dot-alert-confirm';
+
+@Component({
+    selector: 'dot-apps-configuration-detail',
+    templateUrl: './dot-apps-configuration-detail.component.html',
+    styleUrls: ['./dot-apps-configuration-detail.component.scss']
+})
+export class DotAppsConfigurationDetailComponent implements OnInit {
+    @ViewChild('searchInput')
+    messagesKey: { [key: string]: string } = {};
+    apps: DotApps;
+
+    formData: { [key: string]: string };
+    formFields: any[];
+    formValid = false;
+
+    constructor(
+        private route: ActivatedRoute,
+        private dotRouterService: DotRouterService,
+        private dotAppsService: DotAppsService,
+        private dotDialogService: DotAlertConfirmService
+    ) {}
+
+    ngOnInit() {
+        this.route.data
+            .pipe(pluck('data'), take(1))
+            .subscribe(({ messages, app }: AppsResolverData) => {
+                this.apps = app;
+                this.formFields = app.sites[0].secrets;
+                this.messagesKey = messages;
+            });
+    }
+
+    onSubmit() {
+        this.dotAppsService
+            .saveSiteConfiguration(
+                this.apps.key,
+                this.apps.sites[0].id,
+                this.getTransformedFormData()
+            )
+            .subscribe(() => {
```

Comment:
`take1`?

Reply:
I think the take(1) is ok, but it should go into the service (I think)